azure/configuration/az-resource-types.ps1

```diff
@@ -1,12 +1,13 @@
-Class AzureResourceTypes{
-    static [AzureResourceType]$KeyVault = @{
+#Todo: How do we allow overrides of configuration from deployment?
+Class AzResourceTypes{
+    static [AzResourceType]$KeyVault = @{
         Name = "Key Vault"
         Abbreviation = "KV"
         DiagnosticCategoriesToLog = "AuditEvent"
```

Comment:
Feel like eventually want to push DiagnosticCategoriesToLog and MetricCategoriesToLog somewhere else.  Like they a different category of configuration that maybe gets pulled from a 'configuration state', wheras the others are more internal to PF.   Abbreviation kind of straddles types, if wanted to give chance to change

Reply:
> Feel like eventually want to push DiagnosticCategoriesToLog and MetricCategoriesToLog somewhere else. Like they a different category of configuration that maybe gets pulled from a 'configuration state', wheras the others are more internal to PF. Abbreviation kind of straddles types, if wanted to give chance to change

So this is just for the "default" enterprise settings.  Meaning we create a resource in our deployment scripts with no options passed through.  We need to have those "default resource settings" somewhere and this is the start of that.  However, for logging defaults, I agree maybe we just add these as default "options" for the resource instead?  I'll look into that a bit more.  Maybe you are right also, that we put the abbreviation as an option as well?  That way, no option object provided then the defaults are there, but once an option object is instantiated, it is easy to override?  This just means that you would have to override abbreviations/logging defaults every resource then.  I'll think about this more...